src/handlers/shared/check-assignments.ts

```diff
@@ -0,0 +1,65 @@
+import { Context, IssueEvent } from "../../types";
+
+export async function checkPreviousAssignments(context: Context, sender: Context["payload"]["sender"]): Promise<boolean> {
+  const events = await getAssignmentEvents(context);
+  const senderLogin = sender.login.toLowerCase();
+  const userAssignments = events.filter((event) => event.assignee?.toLowerCase() === senderLogin);
+
+  if (userAssignments.length === 0) {
+    return false;
+  }
+
+  const unassignedEvents = userAssignments.filter((event) => event.event === "unassigned");
+  const botUnassigned = unassignedEvents.filter((event) => event.actorType === "Bot");
+  const adminUnassigned = unassignedEvents.filter((event) => event.actor?.toLowerCase() !== senderLogin && event.actorType === "User");
+  const userSelfUnassignViaUi = unassignedEvents.filter((event) => event.actor?.toLowerCase() === senderLogin);
```

Comment:
if an user self unassigns I don't think we should stop him from assigning again.